src/dmtcpplugin.cpp

```diff
@@ -169,10 +169,8 @@ dmtcp_get_tmpdir(void)
 EXTERNC const char *
 dmtcp_get_ckpt_dir()
 {
-  static string tmpdir;
-
-  tmpdir = ProcessInfo::instance().getCkptDir();
-  return tmpdir.c_str();
+  static string *tmpdir = new string(ProcessInfo::instance().getCkptDir());
```

Comment:
Doesn't this create a memory leak by allocating in the heap, instead of tmpdir on the stack?

Reply:
It will only be allocated once because the pointer is declared static and will follow the construct-on-first-use (or initialize-on-first-use) idiom. The initialization will take place upon first entry into the function.